src/darker/config.py

```diff
@@ -0,0 +1,62 @@
+"""Load and save configuration in TOML format"""
+
+import logging
+from argparse import ArgumentParser, Namespace
+from typing import Dict, List, Union, cast
+
+import toml
+from black import find_project_root
+
+
+class TomlArrayLinesEncoder(toml.TomlEncoder):  # type: ignore[name-defined]
+    """Format TOML so list items are each on their own line"""
+
+    def dump_list(self, v: List[str]) -> str:
+        """Format a list value"""
+        return "[{}\n]".format(
+            "".join("\n    {}".format(self.dump_value(item)) for item in v)
+        )
+
+
+DarkerConfig = Dict[str, Union[str, bool, List[str]]]
+
+
+def replace_log_level_name(config: DarkerConfig) -> None:
+    """Replace numeric log level in configuration with the name of the log level"""
+    if "log_level" in config:
+        config["log_level"] = logging.getLevelName(cast(int, config["log_level"]))
+
+
+def load_config() -> DarkerConfig:
+    """Find and load Darker configuration from given path or pyproject.toml"""
+    path = find_project_root((".",)) / "pyproject.toml"
```

Comment:
Is it confusing that Darker configuration is always read from `pyproject.toml` of the repository, while the `--config` command line option only affects where Black and isort configuration is read from?

Reply:
No I think that is fine.

I'm just worried of loading config always wrt `.`, especially from within editors CWD/. may not be what you believe.

Reply:
Ah, I suppose you're suggesting that whenever at least one file/directory is given as an argument, the configuration should be searched for going up the directory hierarchy from there?

Reply:
Yes.

Reply:
I'm also worried about what happens w/o arguments. I think that if no argument get passed you likely don't want to _do_ anything and print help. I think it's safer.

My Home is a git repository, so for example running `darker` without a path from within VIM would literally try to reformat all the files in my home directory ...

Reply:
>running darker without a path from within VIM would literally try to reformat all the files in my home directory

...but only if you specified `src = ["."]` in your `$HOME/pyproject.toml` file.

Reply:
> > running darker without a path from within VIM would literally try to reformat all the files in my home directory
> 
> ...but only if you specified `src = ["."]` in your `$HOME/pyproject.toml` file.

@Carreau I hope this satisfies your concern – I'm proceeding to merging this pull request soon.